packages/automerge-repo/src/DocHandle.ts

```diff
@@ -42,6 +42,9 @@ export class DocHandle<T> //
   #timeoutDelay: number
   #remoteHeads: Record<StorageId, A.Heads> = {}
 
+  // Reference to global meta data that is set on the repo to be attached to each change
+  #globalMetadataRef?: ChangeMetadataRef
```

Comment:
It might be worth allowing this to a be a function `(DocumentId) -> Record<..>`?

Reply:
Yes I think that would be a better API